include/clad/Differentiator/VisitorBase.h

```diff
@@ -191,8 +191,13 @@ namespace clad {
       V.beginBlock();
       func();
       clang::CompoundStmt* body = V.endBlock();
-      clang::Expr* lambda =
-          S.ActOnLambdaExpr(noLoc, body, V.getCurrentScope()).get();
+      #if CLANG_VERSION_MAJOR > 17
+        clang::Expr* lambda =
+            S.ActOnLambdaExpr(noLoc, body).get();
+      #else
+        clang::Expr* lambda =
+            S.ActOnLambdaExpr(noLoc, body, V.getCurrentScope()).get();
+      #endif // CLANG_VERSION_MAJOR > 17
```

Comment:
The idea of Compatibility.h is to isolate all version-related parts of the code in one place. Thus, there are dozens of places where such a conditional compilation is not to be found.

This code should be properly ported to Compatilbility.h. each this is a bit more difficult than a simple "#ifdef", but it pays off.

Reply:
@alexander-penev Could you make a suggestion on how to move this to Compatibility.h ? I made an attempt to move it, but even if it had worked I don't think it was the right way to do it.

Reply:
We could add a clad_compat::ActOnLambdaExpr(S, loc, body, scope) function, but this may lead to unnecessary additional calculation of the last parameter in cases of clang>17. I recommend another approach:
```suggestion
        clang::Expr* lambda = S.ActOnLambdaExpr(noLoc, body/*,*/
          CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V)).get();
```
In Compatibility.h must add this:
```
      #if CLANG_VERSION_MAJOR > 17
        #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) /**/
      #else
        #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) ,(V).getCurrentScope()
      #endif // CLANG_VERSION_MAJOR > 17
```